.github/workflows/ci.yml

```diff
@@ -14,14 +14,22 @@ jobs:
   test:
     runs-on: ubuntu-latest
     strategy:
+      fail-fast: false
       matrix:
-        rust: [stable, beta]
+        # Version should be at least 1.43 because that is when f64::EPSILON
+        # was introduced, which libm depends on.
+        # Next, anything lower than 1.46 was unable to update the registry in
+        # GitHub Actions.
+        toolchain: ['1.46', 'stable', 'beta']
     steps:
-      - uses: actions/checkout@v4
-      - uses: dtolnay/rust-toolchain@master
-        with:
-          toolchain: ${{ matrix.rust }}
-      - uses: Swatinem/rust-cache@v2
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - name: Install Rust
+        run: |
+          rustup update ${{ matrix.toolchain }}
+          rustup default ${{ matrix.toolchain }}
```

Comment:
Using `rustup` instead of `dtolnay/rust-toolchain` because `libm` also uses `rustup` directly.